### DISCUSSION TITLE
Split structures according to atom

### DISCUSSION THREAD

**Discussion by TheophileGaudin** [rdkit/rdkit#5789] [Q&A] [answered]
Hi,

I wonder whether rdkit's Python library includes a simple functionality where you provide an atom index and a molecular object, and it splits the structure into separate substructures by removing that atom (of course for example if this is a terminal atom then it will split a single structure without that atom)?

I browsed the help but I couldn't find the answer myself.

Thanks in advance,
Best regards,
Théophile.

**Comment by bp-kelley** [comment, marked as answer]
To remove a single atom, you should use a read write molecule, aka RWMol which behaves like an ordinary molecule, but adds methods such as:

```
 |  RemoveAtom(...)
 |      RemoveAtom( (RWMol)arg1, (int)arg2) -> None :
 |          Remove the specified atom from the molecule
 |      
 |          C++ signature :
 |              void RemoveAtom(RDKit::ReadWriteMol {lvalue},unsigned int)
 |  
 |  RemoveBond(...)
 |      RemoveBond( (RWMol)arg1, (int)arg2, (int)arg3) -> None :
 |          Remove the specified bond from the molecule
 |      
 |          C++ signature :
 |              void RemoveBond(RDKit::ReadWriteMol {lvalue},unsigned int,unsigned int)
```

for example:

```
from rdkit import Chem
mol = Chem.MolFromSmiles("C1CC1CC2CC2")
rwmol = Chem.RWMol(mol)
rwmol.RemoveAtom(3)
print(Chem.MolToSmiles(rwmol))
```

outputs

```
C1CC1.C1CC1
```

**Reply by TheophileGaudin** [reply]
@bp-kelley Thanks for the answer. I continued on with this however it seems that the indices aren't preserved in the exact order as they appeared on the molecule. In my algorithm I first split according to bond and then according to atom. In the example below I expect that the atom that participates in the bond is removed, ie the C atom that bears 3 methyl groups. Instead one of the methyls is removed.

How do I make it so the indices are preserved and I get the expected result?

See below:

```
from rdkit import Chem
smiles = 'CCC(C)(C)C'
m_noH = Chem.MolFromSmiles(smiles)
bond_to_split = (1,2)
mol_bondcut = Chem.RWMol(m_noH)
indices_cutbond = Chem.GetMolFrags(mol_bondcut)
frag2_rw.RemoveAtom(indices_cutbond[1].index(bond_to_split[1])) 
smiles_frag2 = [Chem.MolToSmiles(m_ind) for m_ind in Chem.GetMolFrags(frag2_rw, asMols=True)]
print(smiles_frag2)
```

Result:
`['CCC']`
Expected result:
`['C','C','C']`

**Reply by TheophileGaudin** [reply]
Self-reply. I fixed this by explicitly preventing sanitization and canonicalization:
```
from rdkit import Chem
smiles = 'CCC(C)(C)C'
m_noH = Chem.MolFromSmiles(smiles)
bond_to_split = (1,2)
mol_bondcut = Chem.RWMol(m_noH)
indices_cutbond = Chem.GetMolFrags(mol_bondcut)
frag2_rw.RemoveAtom(indices_cutbond[1].index(bond_to_split[1])) 
smiles_frag2 = [Chem.MolToSmiles(m_ind, canonical=False) for m_ind in Chem.GetMolFrags(frag2_rw, asMols=True, sanitizeFrags = False)]
print(smiles_frag2)
```
Result as expected:
`['C','C','C']`
